### PR TITLE
Member data dump button

### PR DESCRIPTION
**Pull request by Thomas-Malcolm**
Added a button which dumps an excel spreadsheet (secretly just a csv) containing member data to be submitted to UQU (or for other purposes I guess)

### CODE REVIEW

File: uqcs/admin.py [hunk 1]
```diff
@@ -151,6 +152,24 @@ def delete(s, admin_user, member_id):
...
+
+    for mem in members:
+        fn = mem.first_name
+        ln = mem.last_name
+        em = mem.email
+        data.write(",".join([fn, ln, em]).encode('utf-8') + b"\r\n")
+    
+    data.seek(0)
+
+    return send_file(data, as_attachment=True, download_name="member_list.xlsx")
```
Comment (JamesDearlove) [inline uqcs/admin.py]: I'd keep this as a `.csv`, they never harmed anyone

File: uqcs/admin.py [hunk 2]
```diff
@@ -151,6 +152,24 @@ def delete(s, admin_user, member_id):
...
+def dump_members(s, admin_user):
+
+    members = s.query(m.Member).filter(m.Member.paid != None).all()
+    data = io.BytesIO()
+
+    for mem in members:
+        fn = mem.first_name
+        ln = mem.last_name
+        em = mem.email
+        data.write(",".join([fn, ln, em]).encode('utf-8') + b"\r\n")
```
Comment (JamesDearlove) [inline uqcs/admin.py]: Union also ask for student numbers, so we should include that (and also indicate non students)

**Review by JamesDearlove — COMMENTED** [review]
Looks good, I no longer have anything setup to test but the code looks good. Just a couple of comments

File: uqcs/admin.py (hunk 1 above)
Comment (lindenwells) [inline uqcs/admin.py]:
```suggestion
    date = str(dt.today())
    return send_file(data, as_attachment=True, download_name=f"member_list_{date}.csv")
```

Comment (Thomas-Malcolm) [reply uqcs/admin.py]: `.csv` hate crew, where are you (yeah I'll change it over)

File: uqcs/admin.py (hunk 2 above)
Comment (Thomas-Malcolm) [reply uqcs/admin.py]: Good point, I'll add that in

File: uqcs/admin.py (hunk 1 above)
Comment (Thomas-Malcolm) [reply uqcs/admin.py]: Not a bad idea, will also add this in

**Review by JamesDearlove — APPROVED** [review]
Assuming this works (i dont have a testing environment), lgtm

**Pull request merged**